hydromt/raster.py

```
@@ -262,53 +267,66 @@ def crs(self) -> CRS:
+            raise ValueError(f"Invalid CRS type: {type(input_crs)}")
         else:
+            crs = None
             for name in crs_names:
-                # check default > GEO_MAP_COORDS attrs
-                crs = self._obj.coords[GEO_MAP_COORD].attrs.get(name, None)
-                if crs is None:  # global attrs
-                    crs = self._obj.attrs.pop(name, None)
-                for var in names:  # data var and coords attrs
-                    if name in self._obj[var].attrs:
-                        crs = self._obj[var].attrs.pop(name)
-                        break
-                if crs is not None:
-                    # avoid Warning 1: +init=epsg:XXXX syntax is deprecated
-                    crs = crs.removeprefix("+init=") if isinstance(crs, str) else crs
-                    try:
-                        input_crs = pyproj.CRS.from_user_input(crs)
-                        break
-                    except RuntimeError:
-                        pass
+                # check default > GEO_MAP_COORDS attrs, then global attrs
+                if name in self.attrs:
+                    crs = self.attrs.get(name)
+                    break
+                if name in self._obj.attrs:
+                    crs = self._obj.attrs.pop(name)
+                    break
+            if crs is None:  # check data var and coords attrs
+                for var in names:
+                    for name in crs_names:
+                        if name in self._obj[var].attrs:
+                            crs = self._obj[var].attrs.pop(name)
+                            break
+            if crs is not None:
+                # avoid Warning 1: +init=epsg:XXXX syntax is deprecated
+                if isinstance(crs, str):
+                    crs = crs.removeprefix("+init=")
+                try:
+                    input_crs = pyproj.CRS.from_user_input(crs)
+                except RuntimeError:
+                    pass
```

Maybe we should also raise an error here? otherwise might be good to add a comment that this is intentional